runtime/compiler/p/codegen/PPCJNILinkage.cpp

```diff
@@ -219,7 +219,12 @@ TR::Register *J9::Power::JNILinkage::buildDirectDispatch(TR::Node *callNode)
       {
       // We need to kill all the non-volatiles so that they'll be in a stack frame in case
       // gc needs to find them.
-      if (comp()->target().is32Bit())
+      if (comp()->target().is64Bit())
```

Comment:
Here and in a few other places, we don't check the disableTOC option. Is that by design?

Reply:
yes, that is by design, since disableTOC carries more meanings than POWER10.  For example, on POWER8/9, AOT also disabled pTOC (but gr16 is not free in those cases).  gr16 is only free on P10 and later (when pTOC is disabled as well).

Reply:
Sorry, I don't completely follow: will we be able to run with pTOC on P10?

Reply:
after this PR is merged ... no, pTOC is completely gone on P10 and gr16 is recovered (you cannot reference to pTOC anymore on P10).  before this PR is merged, gr16 is still reserved and pTOC can be referenced and used in theory (but disabled in reality already).

Reply:
Thanks. Just to clarify after an off-line discussion: it was a design decision to disable pTOC completely and always use gr16 on P10.